eox_nelp/signals/tests/test_receivers.py

```diff
@@ -64,3 +72,173 @@ def test_call_dispatch(self, dispatch_mock):
+        """Test that the asynchronous task wont' be called when the setting is not active.
+
+        Expected behavior:
+            - create_external_certificate is not called
+        """
+        certificate_publisher(self.certificate_data, self.metadata)
+
+        mock_task.delay.assert_not_called()
+
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_invalid_mode(self, mock_task):
+        """Test when the certificate data has an invalid mode.
+
+        Expected behavior:
+            - create_external_certificate is not called.
+            - Invalid error was logged.
+        """
+        invalid_mode = "audit"
+        log_info = (
+            f"The {invalid_mode} certificate associated with the user <{self.username}> and course <{self.course_key}> "
+            "doesn't have a valid mode and therefore its data won't be published."
+        )
+
+        certificate_data = CertificateData(
+            user=UserData(
+                pii=UserPersonalData(
+                    username=self.username,
+                    email="harry@potter.com",
+                    name="Harry Potter",
+                ),
+                id=10,
+                is_active=True,
+            ),
+            course=CourseData(
+                course_key=self.course_key,
+            ),
+            mode=invalid_mode,
+            grade=5,
+            current_status="downloadable",
+            download_url="",
+            name="",
+        )
+
+        with self.assertLogs(receivers.__name__, level="INFO") as logs:
+            certificate_publisher(certificate_data, self.metadata)
+
+        mock_task.delay.assert_not_called()
+        self.assertEqual(logs.output, [
+            f"INFO:{receivers.__name__}:{log_info}"
+        ])
+
+    @patch("eox_nelp.signals.receivers._generate_external_certificate_data")
+    @patch("eox_nelp.signals.receivers.create_external_certificate")
+    def test_create_call(self, mock_task, generate_data_mock):
+        """Test when the certificate mode is valid and the asynchronous task is called
+
+        Expected behavior:
+            - _generate_external_certificate_data is called with the right parameters.
+            - create_external_certificate is called with the _generate_external_certificate_data output.
+            - Info was logged.
+        """
+        generate_data_mock.return_value = {
+            "test": True,
+        }
+        log_info = (
+            f"The no-id-professional certificate associated with the user <{self.username}> and "
+            f"course <{self.course_key}> has been already generated and its data will be sent "
+            "to the NELC certificate service."
+        )
+
+        with self.assertLogs(receivers.__name__, level="INFO") as logs:
+            certificate_publisher(self.certificate_data, self.metadata)
+
+        generate_data_mock.assert_called_with(
+            timestamp=self.metadata.time,
```

Comment:
the time of the object where is generated, or is auto generated?
https://github.com/eduNEXT/eox-nelp/blob/2b62c92f911e8638b981a4ea432ec490f9a1749a/eox_nelp/signals/tests/test_receivers.py#L103-L106

Reply:
https://github.com/eduNEXT/openedx-events/blob/main/openedx_events/data.py#L42